### PR TITLE
I have added filter options to see events virtual/in person

### LINKED ISSUE #2528: [FEATURE] Filter events for virtual / in person
### Description

After this PR is merged https://github.com/EddieHubCommunity/LinkFree/pull/2527

Click the key cards to filter events

Screenshot 2022-12-28 at 22 09 27

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by abhijain2003**
## Fixes Issue

#2528 

## Changes proposed

-> initialize state variable in events.js file with toggle values "virtual"/"person"
-> passing setState method in eventKey.js file as a props so I can toggle values on button clicks
-> implementing filter method before map in events.js file based on value of state variable
-> added a scale-105 class on hover in EventKey.js file

## Screenshots

https://user-images.githubusercontent.com/91718608/210301611-dd010e99-335f-480d-92a9-23702ee60537.mp4

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/2605"></a>

### CODE REVIEW

File: components/event/EventKey.js [hunk 1]
```diff
@@ -1,10 +1,15 @@
 import { IconContext } from "react-icons";
 import { MdOutlineOnlinePrediction, MdOutlinePeople } from "react-icons/md";
 
-export default function EventKey() {
+export default function EventKey({ onChange }) {
+
+  function handleChangeType(value) {
```
Comment (eddiejaoude) [inline components/event/EventKey.js:6]: I don't think this is needed, `onChange` can be called directly

File: components/event/EventKey.js [hunk 2]
```diff
@@ -1,10 +1,15 @@
 import { IconContext } from "react-icons";
 import { MdOutlineOnlinePrediction, MdOutlinePeople } from "react-icons/md";
 
-export default function EventKey() {
+export default function EventKey({ onChange }) {
```
Comment (eddiejaoude) [inline components/event/EventKey.js:4]: I think `onChange` should be named more descriptively

File: components/user/UserEvents.js [hunk 3]
```diff
@@ -1,17 +1,26 @@
 import EventCard from "../event/EventCard";
 import Alert from "../Alert";
 import EventKey from "../event/EventKey";
+import { useState } from "react";
```
Comment (eddiejaoude) [inline components/user/UserEvents.js:4]: This doesn't seem to be used?

File: components/user/UserEvents.js [hunk 4]
```diff
@@ -1,17 +1,26 @@
...
+          data.events
+          .map((event, index) => {
+            console.log(event)
+            return (
+              (
             <EventCard event={event} key={index} />
-          ))}
+          )
+            )
+          })}
```
Comment (eddiejaoude) [inline components/user/UserEvents.js:23]: These changes seem to be the same but with a console log?

File: pages/events.js [hunk 5]
```diff
@@ -20,6 +21,8 @@ export async function getServerSideProps(context) {
 }
 
 export default function Events({ events }) {
+  console.log('events', events)
```
Comment (eddiejaoude) [inline pages/events.js:24]: Debug code should be removed

File: pages/events.js [hunk 6]
```diff
@@ -20,6 +21,8 @@ export async function getServerSideProps(context) {
...
+  const [eventType, seteventType] = useState('virtual');
```
Comment (eddiejaoude) [inline pages/events.js:25]: By default it should not be filtered and should show all events

File: pages/events.js [hunk 7]
```diff
@@ -30,11 +33,19 @@ export default function Events({ events }) {
...
         <h1 className="text-4xl mb-4 font-bold">Community events</h1>
 
-        <EventKey />
+        <EventKey onChange={(newValue) => seteventType(newValue)} />
 
         {!events.length && <Alert type="info" message="No events found" />}
         <ul role="list" className="divide-y divide-gray-200">
-          {events.map((event, index) => (
+          {events
+          .filter((itm, _) => {
```
Comment (eddiejaoude) [inline pages/events.js:41]: what is the `_` for?

**Comment by abhijain2003** [comment]
can you give me one more chance I'll resolve all conversation

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you for the contribution 👍 

- I think the `package-lock` file does not need to be changed
- I left some inline comments

**Comment by eddiejaoude** [comment]
> can you give me one more chance I'll resolve all conversation

of course, we are here to help, any questions let us know 👍

File: components/user/UserEvents.js (hunk 3 above)
Comment (abhijain2003) [reply components/user/UserEvents.js:4]: I didn't get you can it be done without useState variables

Comment (eddiejaoude) [reply components/user/UserEvents.js:4]: This import line is added to the file but I don't see this import used in the file?

**Comment by abhijain2003** [comment]
can I add a small button to take back to previous stage and show all events

**Comment by abhijain2003** [comment]
I have refactor the code
remove useless function in EventKey.js and calling onChange Directly
renamed onChange to onToggleEventType
renamed child inside filter method from itm to event
replaced _ with index
displaying all events by default 

didn't made any changes in package-lock previously so didn't toched it now as well
I don't think it can be done without useState variable so, I did not changed it

Can I commit it or would you like to give me more advices

**Comment by eddiejaoude** [comment]
Great! Always commit otherwise we can't see the changes - GitHub will append them to this PR for review

**Comment by ChinmayMhatre** [comment]
> I have refactor the code remove useless function in EventKey.js and calling onChange Directly renamed onChange to onToggleEventType renamed child inside filter method from itm to event replaced _ with index displaying all events by default
> 
> didn't made any changes in package-lock previously so didn't toched it now as well I don't think it can be done without useState variable so, I did not changed it
> 
> Can I commit it or would you like to give me more advices

Hey @abhijain2003 , I think what Eddie means. By the `useState`  is that the import isn't used in `UserEvent.js` so you could remove it. And Also that the `console.log` should be removed from everywhere after you are done debugging.

**Comment by abhijain2003** [comment]
ok
I just solved all issues hope this is ready to merge now

**Comment by ChinmayMhatre** [comment]
I think we should add a border to the selected event type. What do you think @abhijain2003 , @eddiejaoude ?

**Comment by ChinmayMhatre** [comment]
> ok I just solved all issues hope this is ready to merge now

Hey , I think eddie also had suggested to remove the package-lock.json from the pr.

**Comment by abhijain2003** [comment]
> I think we should add a border to the selected event type. What do you think @abhijain2003 , @eddiejaoude ?

I added a kind of scale property on hover on selected button

**Comment by eddiejaoude** [comment]
> I think we should add a border to the selected event type

I think this can come in another PR

**Comment by abhijain2003** [comment]
so will you merge it now

**Comment by eddiejaoude** [comment]
> so will you merge it now

@abhijain2003 please review remaining inline comments

**Comment by abhijain2003** [comment]
ok eddie

**Comment by abhijain2003** [comment]
eddie as you told me I have removed package-lock file and made a new commit 

hope there are no further checks are remaining

**Comment by eddiejaoude** [comment]
> eddie as you told me I have removed package-lock file and made a new commit

Please do not remove this file, it is important, I asked to remove your changes to the file. You can use `git checkout main --- package-lock.json` and this should return the file to the state before your changes.

Also please address the remaining 2 inline comments that are not in a resolved state yet (these are the ones that are still expanded on this page (the others are collapsed as they are resolved)

**Comment by abhijain2003** [comment]
hey Eddie,  I think I messed up the contribution, but I resolved all conversations now, 

Thanks for Helping

**Comment by eddiejaoude** [comment]
@abhijain2003 you marked this as resolved without a comment or changing the lines of code it is referring to, this makes it difficult to review. Do you need some help with this? I have marked it back to unresolved.

Screenshot 2023-01-04 at 09 53 13

**Comment by abhijain2003** [comment]
not at all eddie this button was by mistaken pressed I am sorry for that,

and I made changes in code, removed unused imports and this issue is also resolved

**Comment by abhijain2003** [comment]
done eddie , i commited to my branch all changes

hope now it is resolved

File: pages/events.js [hunk 8]
```diff
@@ -20,6 +21,8 @@ export async function getServerSideProps(context) {
 }
 
 export default function Events({ events }) {
+
+  const [eventType, seteventType] = useState('');
```
Comment (eddiejaoude) [inline pages/events.js:25]: `useState` will need an import

File: pages/events.js [hunk 9]
```diff
@@ -30,13 +33,23 @@ export default function Events({ events }) {
...
-        <EventKey />
+        <EventKey onToggleEventType={(newValue) => seteventType(newValue)} />
 
         {!events.length && <Alert type="info" message="No events found" />}
         <ul role="list" className="divide-y divide-gray-200">
-          {events.map((event, index) => (
-            <EventCard event={event} username={event.username} key={index} />
-          ))}
+          {events
+            .filter((event, index) => {
```
Comment (eddiejaoude) [inline pages/events.js:42]: I don't think `index` is used here?

File: pages/events.js (hunk 8 above)
Comment (abhijain2003) [reply pages/events.js:26]: thank eddie, for letting me know I just added it

File: pages/events.js (hunk 9 above)
Comment (abhijain2003) [reply pages/events.js]: yes you are right, I have removed it

**Comment by eddiejaoude** [comment]
Thank you 👍  I will resolve conflicts

**Comment by abhijain2003** [comment]
thanks to you, you really helped me in my initial stage of open source looking forward to contribute more

**Comment by eddiejaoude** [comment]
Thank you for your hard work @abhijain2003 💪

**Pull request merged**